iceoryx_hoofs/posix/filesystem/source/file.cpp

```diff
@@ -0,0 +1,438 @@
+    case EBUSY:
+        IOX_LOG(ERROR) << "Unable to remove file since it is currently in use.";
+        return iox::error<FileRemoveError>(FileRemoveError::CurrentlyInUse);
+    case EIO:
+        IOX_LOG(ERROR) << "Unable to remove file due to an IO failure.";
+        return iox::error<FileRemoveError>(FileRemoveError::IoFailure);
+    case ELOOP:
+        IOX_LOG(ERROR) << "Unable to remove file due to too many symbolic links.";
+        return iox::error<FileRemoveError>(FileRemoveError::TooManySymbolicLinksEncountered);
+    case ENOMEM:
+        IOX_LOG(ERROR) << "Unable to remove file due to insufficient kernel memory.";
+        return iox::error<FileRemoveError>(FileRemoveError::InsufficientKernelMemory);
+    case EISDIR:
+        IOX_LOG(ERROR) << "Unable to remove file since it is a directory.";
+        return iox::error<FileRemoveError>(FileRemoveError::IsDirectory);
+    case EROFS:
+        IOX_LOG(ERROR) << "Unable to remove file since it resides on a read-only file system.";
+        return iox::error<FileRemoveError>(FileRemoveError::ReadOnlyFilesystem);
+    default:
+        IOX_LOG(ERROR) << "Unable to remove file since an unknown error occurred.";
+        return iox::error<FileRemoveError>(FileRemoveError::UnknownError);
+    }
+}
+
+expected<FileOffsetError> File::set_offset(const uint64_t offset) const noexcept
+{
+    auto result = posix::posixCall(iox_lseek)(m_file_descriptor, static_cast<iox_off_t>(offset), IOX_SEEK_SET)
+                      .failureReturnValue(-1)
+                      .evaluate();
+
+    if (!result.has_error())
+    {
+        if (result->value == static_cast<iox_off_t>(offset))
+        {
+            return iox::success<>();
+        }
+
+        IOX_LOG(ERROR) << "Unable to set file offset position since it set to the wrong offset position.";
+        return iox::error<FileOffsetError>(FileOffsetError::OffsetAtWrongPosition);
+    }
+
+
+    switch (result.get_error().errnum)
+    {
+    case EINVAL:
+        IOX_FALLTHROUGH;
+    case ENXIO:
+        IOX_LOG(ERROR) << "Unable to set file offset position since it beyond the file limits.";
+        return iox::error<FileOffsetError>(FileOffsetError::OffsetBeyondFileLimits);
+    case EOVERFLOW:
+        IOX_LOG(ERROR)
+            << "Unable to set file offset position since the file is too large and the offset would overflow.";
+        return iox::error<FileOffsetError>(FileOffsetError::FileOffsetOverflow);
+    case EPIPE:
+        IOX_LOG(ERROR) << "Unable to set file offset position since seeking is not supported by the file type.";
+        return iox::error<FileOffsetError>(FileOffsetError::SeekingNotSupportedByFileType);
+    default:
+        IOX_LOG(ERROR) << "Unable to remove file since an unknown error occurred.";
+        return iox::error<FileOffsetError>(FileOffsetError::UnknownError);
+    }
+}
+
+expected<uint64_t, FileReadError> File::read(uint8_t* const buffer, const uint64_t buffer_len) const noexcept
```

Comment:
This API is a bit error prone but I currently also don't have a better solution

Reply:
I nice slice of memory would be great ... but this can be something for the near future.